### PR TITLE
Added org.junit.platform.commons.support.ClassSupport

### PR DESCRIPTION
**Pull request by jlink**
## Overview
Addresses https://github.com/junit-team/junit5/issues/1144

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests)
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/meta/API.html)
- [ ] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [ ] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: junit-platform-commons/src/main/java/org/junit/platform/commons/support/ClassSupport.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import java.util.function.Function;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.util.ClassUtils;
+
+/**
+ * Common class support.
+ *
+ * @since 1.0.2
```
Comment (marcphilipp) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/support/ClassSupport.java]: We'll release this in 1.1.

File: junit-platform-commons/src/main/java/org/junit/platform/commons/support/ClassSupport.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ */
+@API(status = MAINTAINED, since = "5.0.2")
```
Comment (marcphilipp) [inline junit-platform-commons/src/main/java/org/junit/platform/commons/support/ClassSupport.java]: Ditto.

File: platform-tests/src/test/java/org/junit/platform/commons/support/ClassSupportTests.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.util.ClassUtils;
+
+/**
+ * @since 1.0.2
```
Comment (marcphilipp) [inline platform-tests/src/test/java/org/junit/platform/commons/support/ClassSupportTests.java]: Ditto.

**Comment by jlink** [comment]
gradle checkStyleTest did not show any formatting violations here.

**Comment by sbrannen** [comment]
There are still Spotless violations: 

https://ci.appveyor.com/project/marcphilipp/junit5/build/2016#L1339

_Run `gradlew spotlessApply` to fix these violations._ 😉

**Comment by sbrannen** [comment]
This has been merged into `master`. Thanks!

**Pull request merged**